session.go

```diff
@@ -846,7 +869,11 @@ func (s *session) Txn() kv.Transaction {
 
 func (s *session) NewTxn() error {
 	if s.txn != nil && s.txn.Valid() {
-		err := s.CommitTxn()
+		ctx := s.goCtx
+		if ctx == nil {
```

Comment:
When will `ctx` be nil?

Reply:
Normal cases won't go here, but in the testing `ctx` maybe not set.